tests/lib/test-common.bash.in

```diff
@@ -92,3 +78,36 @@ function get_average(){
 function random_name() {
 	echo $(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
 }
+
+function check_active_process() {
+	process=$1
+	if pgrep -f "$process" > /dev/null; then
+		return 1
+	else
+		return 0
+	fi
+}
+
+# This function checks if there are containers or
+# cc-shim and qemu-lite processes up, if found, they are
+# killed to start test with clean environment.
+# The function is intended to be used in the setup() of
+# the docker integration tests for cc-oci-runtime.
+function kill_processes_before_start() {
+	run bash -c "[ $($DOCKER_EXE ps -q) ] && $DOCKER_EXE kill $($DOCKER_EXE ps -q)"
+	run bash -c "check_active_process $HYPERVISOR_PATH || killall $HYPERVISOR_PATH"
+	run bash -c "check_active_process $CC_SHIM_PATH || killall $CC_SHIM_PATH"
+
+}
+
+# This function checks for active processes that were
+# left behind by cc-oci-runtime.
+# The function is intended to be used in the teardown() of
+# the docker integration tests for cc-oci-runtime.
+function check_no_processes_up() {
+	# Sleep 1 second to wait for cc-shim and qemu to end properly
+	sleep 1
```

Comment:
I don't think this sleep is going to be reliable. If we must do this, then a polling loop would be better where the loop checks for the processes for -- say -- up to 60 seconds and prints out details (atleast a count) on the remaining processes every second. If after that timeout period there are still processes running, I'd print the full details (list of pids and maybe even the contents of `/proc/$pid/stack` for each) and then error out.

Admittedly, 60 seconds is a total guess, but rather than waiting for what we think is the *minimum* amount of time (fragile - highly likely we'll need to keep on increasing it), we're waiting for *up to* what we think is the *maximum* reasonable amount of time (*hopefully* will never need to be changed). And if after that maximum time processes still exist, we can identify more easily what the problem is.